### PR TITLE
Display Used Hash Key, Maximum RPM and RPM to status webpage *WIP*

### PR DESCRIPTION
**Pull request by Alderon86**
## Info
This is at the moment not usable due to changes in develop and its in the state of getting reworked right now! Be Patiend please! Big Changes inc!
## Description
Check currently used Hash Key, check for Maximum RPM and RPM left for your used Key!
Display it in console and Status Page.
For 100% comparison check your DB its written in there from now on.
Special thx to @hokiepokedad for updated scheduler and @onilton for reviewing code

## How Has This Been Tested?
My Own Map

## Screenshots (if appropriate):
nice

statuslog
success

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.

### CODE REVIEW

**Comment by pbrent84** [comment]
Rip that other hash key

**Comment by Alderon86** [comment]
@pbrent84 what do u mean? second key is not displayed correct?

**Comment by klatterhans** [comment]
Nice one. Does it work with the same key in multiple runserver.py ?

**Comment by Alderon86** [comment]
@klatterhans yes it does work on multiple instances.
yeah changes are comming, dunno why its not showing via upload

**Comment by Alderon86** [comment]
updatet, multiple keys read correct - check
              Maximum RPM Bug - fixed
              database writing for historical comparison - check

**Comment by klatterhans** [comment]
Thanks for updating. Working now!
But... I wonder one thing. Encoutering a Pokemon to see IV doesn't cost any RPM? 
found 5 pokemons...  150 RPM has 145 RPM left.

Should be more or is that the encounter thing? I'm confused

**Comment by Alderon86** [comment]
encoutering is one request yes, as mentioned above, its a little bit delayed, but updatet after every getmapobjects call

**Comment by Alderon86** [comment]
@jaake77 looking into it already

**Comment by Alderon86** [comment]
@Barloklol thank you! fixed it

File: pogom/models.py [hunk 1]
```diff
@@ -968,6 +968,9 @@ class WorkerStatus(BaseModel):
     skip = IntegerField()
     last_modified = DateTimeField(index=True)
     message = CharField(max_length=255)
+    hash_key = CharField(index=True, max_length=50)
```
Comment (Barloklol) [inline pogom/models.py]: Should be changed here as well

Comment (Alderon86) [inline pogom/models.py]: true

**Comment by Alderon86** [comment]
@Barloklol should be ok now!?

**Comment by Barloklol** [comment]
Yea those were the changes I made, works fine

**Comment by hokiepokedad** [comment]
@Alderon86 u may want to rebase, looks like your latest commit overwrites the latest commits on develop

**Comment by Alderon86** [comment]
@hokiepokedad im on it, already noticed the bullshit i did

**Comment by Qualimiox** [comment]
Argh, I just updated and now this doesn't work due to the mismatch and develop doesn't work due to the updated db structure. If I change the version back to 12 I get "Field 'maximum_rpm' doesn't have a default value" errors. Can anyone tell me what columns I need to drop to properly go back to version 12?

**Comment by Alderon86** [comment]
@Qualimiox just wait a few mins, im on it

**Comment by bbdoc** [comment]
Hi @Alderon86 ,

Stupid question regarding this one, but why do you give Hash Key status at worker level ? We usually use 1 or several keys in config file shared for all users (even all instances in my case). Wouldn't it be better to simply give this information on top of status page with a list of the keys and RPM status per key ?

Might be a good idea (but don't know if easy to do) to create like 2 tabs on status page, one for current status and one for RPM status (which btw could give some room for future PRs to tackle status page improvements ;-) ).

**Comment by Alderon86** [comment]
Firstly with this you can track your rpm usage in your database also.
I also thought about creating a header, but since there are at least 2 prs with header code im not doing this one also. There is also a PR which gives you the ability to hide/show worker columns on status page. total rpm usage per worker and maybe some other additions will follow if other prs get merged. dont want to steal code and claim it as my own!

File: pogom/search.py [hunk 2]
```diff
@@ -978,6 +989,27 @@ def search_worker_thread(args, account_queue, account_failures,
...
+                            if (key == key_scheduler.current_key()):
+                                status['hash_key'] = (
+                                    key_scheduler.current_key())
+                                maximum = HashServer.status.get('maximum')
+                                status['maximum_rpm'] = maximum
+                                try:
+                                    request = api.create_request()
+                                    remaining = HashServer.status.get(
+                                        'remaining')
+                                    request.call()
```
Comment (catnet23) [inline pogom/search.py]: So we have to deal with 1 request more after map_request for getting the correct status of the RPM.

Comment (rpotpogo) [inline pogom/search.py]: every 5th call or so ends up with this error for me after applying this pr.
2017-01-27 09:11:00,737 [search-worker-50][        search][   ERROR] Map parse failed at x,y, abandoning location. username may be banned.. Exception message: NianticThrottlingException('Request throttled by server... slow down man',)
Traceback (most recent call last):
  File "C:\PokemonGo-Map\pogom\search.py", line 1012, in search_worker_thread
    request.call()
  File "C:\Python27\lib\site-packages\pgoapi\pgoapi.py", line 264, in call
    response = request.request(self._api_endpoint, self._req_method_list, self.get_position())
  File "C:\Python27\lib\site-packages\pgoapi\rpc_api.py", line 165, in request
    raise NianticThrottlingException("Request throttled by server... slow down man")
NianticThrottlingException: Request throttled by server... slow down man

Comment (Alderon86) [inline pogom/search.py]: extra request got removed @rpotpogo

**Comment by Alderon86** [comment]
Rebased to match #1686 (hopefully)

**Comment by bjodasso** [comment]
I'm struggling to understand how this works. I'm running 175 workers across 7 instances and yet my 500 hash key never goes below 400 rpm left. Is it only showing rpm per instance? because either I'm throttling big time and not realizing it or my 175 accounts just aren't hitting the servers hard.

**Comment by Alderon86** [comment]
@bjodasso initial scan complete? most tth found? then its totally normal, my 150 workers are barely hiiting 200 rpm on spawnscanning

**Comment by bjodasso** [comment]
I had 3 instances of 25 workers still in initial scan and 4 instances complete. So maybe it really isn't using add much rpm as I would expect. Thanks.

**Comment by NHLGames** [comment]
I'd like to compliment this PR in its resiliency against changes in the main branch, so far no merge conflicts 👍

**Comment by Alderon86** [comment]
@NHLGames thanks <3

**Comment by hokiepokedad** [comment]
@Alderon86 Now that I've had some time to take a look at things, I agree with @bbdoc. I'm wondering if we should modify the status page altogether to switch from workers to hash keys.  Hash keys are independent from workers/accounts.  Displaying the hash key(s) status in the header also wouldn't be beneficial since there can be many hash keys involved.  We should have a completely different section to display all keys and their stats. Perhaps a new hashkeys table storing this information and including steps in the database cleaning to remove unused hashkeys from the table.

**Pull request closed without merging**